src/pages/ToDo/ToDoListItem.jsx

```diff
@@ -1,27 +1,40 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
 
-const ToDoListItem = (props) => {
-  const {isComplete,id,onChange,text,handleDropDown,isMenuOpen,onDelete,flip} = props;
-  const isCompleteClass = isComplete ? 'complete' : 'incomplete';
-  const isMenuOpenClass = isMenuOpen ? 'open' : 'closed';
-  const flipAction = flip ? 'flip' : '';
+class ToDoListItem extends React.Component {
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.isComplete && nextProps.isComplete) {
+      window.requestAnimationFrame(() => {
+        this.label.classList.add("flip");
+      });
+    } else if (this.props.isComplete && !nextProps.isComplete) {
+      window.requestAnimationFrame(() => {
+        this.label.classList.remove("flip");
+      });
+    }
+  }
```

Comment:
Programmatically add and remove class from label element based on prop change. This means you don't have to add a new property to the data. This will only run if the task is checked by the user. The animation doesn't run if the task is already complete on load.